src/libraries/Common/src/System/Net/NTAuthentication.Common.cs

```diff
@@ -46,7 +47,7 @@ internal string ClientSpecifiedSpn
                     _clientSpecifiedSpn = GetClientSpecifiedSpn();
                 }
 
-                return _clientSpecifiedSpn;
+                return _clientSpecifiedSpn!;
```

Comment:
`GetClientSpecifiedSpn()` can return `null`. Are we sure this will never be null?

Maybe we can change `NegotiateStreamPal.QueryContextClientSpecifiedSpn` to return a `string` and not a nullable string?